salt/elife-xpub/scripts/setup-database.sh

```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 set -e
 
-if ! docker-compose -f docker-compose.yml -f docker-compose.formula.yml exec -T postgres /bin/bash -c "until echo > /dev/tcp/postgres/5432; do sleep 1; done; psql xpub xpub -c \"SELECT 'public.entities'::regclass\""; then
+if ! docker-compose -f docker-compose.yml -f docker-compose.formula.yml exec -T postgres /bin/bash -c "until echo > /dev/tcp/postgres/5432; do sleep 1; done; psql -c \"SELECT 'public.entities'::regclass\""; then
     docker-compose  -f docker-compose.yml -f docker-compose.formula.yml run app /bin/bash -c "until echo > /dev/tcp/postgres/5432; do sleep 1; done; npx pubsweet setupdb --username={{ pillar.elife_xpub.database.user }} --password={{ pillar.elife_xpub.database.password }} --email={{ pillar.elife_xpub.database.email }}"
```

Comment:
Yes, the entire waiting concept needs to be addresses

I'm not sure if the script actually needs to wait for the following checks:
- Do we rely on the postgres container to be up and running before we can use `psql`?
- Do we rely on the app container to be up and running before we can use `npx pubsweet`?
- Do we need to do any checks to make sure the remote database is accessible?

Reply:
To summarize:

> Do we rely on the postgres container to be up and running before we can use psql

As soon as `docker-compose exec` lets you in, you're fine to use `psql`. No need to always wait for the local `postgres` server. We may need to wait for RDS in rare cases e.g. it's Monday and a testing environment is being turned on, EC2 is faster than RDS and the containers are started too soon.

> Do we rely on the app container to be up and running before we can use npx pubsweet?

You don't need that because this uses another `run` command, so a separate container is created. I think you also may not be able to wait, because this creates the database and the application (used to?) fail without a database.

> Do we need to do any checks to make sure the remote database is accessible?

The existing waiting, with configured host and port, should be enough.

> I think we can be a bit more clever than to wait forever if there is no database :)

We should introduce a timeout later; not trivial to push everything into a single bash command

> What happens if this is using RDS and port 5432 is not local?

It still waits, but for the wrong thing